### DISCUSSION TITLE
Official package types support from the `typescript` package

### DISCUSSION THREAD

**Discussion by trusktr** [meteor/meteor#11829] [Feature Requests]
@edemaine and I chatted about supporting types for package import specifiers like `meteor/foo`.

See this discussion: https://discord.com/channels/287346165211463680/287346165211463680/926240556617715742 (btw, comes to Discord, it's so much better than Slack 😊)

TLDR,

- although we can publish to `@types/meteor` but that's not ideal at all when we'd like to write source (including types) all in once olace,
- or we can publish packages directly to npm but that may not be ideal if a package is for Meteor only
- so we thought why not auto generate a `tsconfig.json` inside `.meteor/local` with correct `paths` config for meteor packages, and have a user's tsconfig.json extend from that using `extends`. Or similar.

The bottom line is that publishing to atmosphere, and having type support for those packages, would not longer be a burden (keeping separate types in sync in the [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped) repo is quite a burden).

**Comment by fredmaiaarantes** [comment]
This is in progress now @trusktr. There is this discussion with more information:

[Typescript integration for Meteor packages](https://github.com/meteor/meteor/discussions/12080)
